### PR TITLE
Add 'poll' function to custreamz kafka consumer

### PR DESCRIPTION
**Pull request by jdye64**
## Description
Streamz has updated their codebase to include a call to the Confluent Kafka Consumer library function 'poll'. Currently custreamz does not include this method. This PR adds the 'poll' function to custreamz to simply proxy the call to the underlying confluent kafka library so that streamz is no longer broken for end users. Without this function end users are no longer able to use custreamz with newer versions of the streamz library.

This closes: #13600

### CODE REVIEW

**Comment by bdice** [comment]
Does this need to target 23.08 or 23.10? Breakage with new versions of streamz sounds like a candidate for 23.08 despite code freeze.

**Comment by jdye64** [comment]
23.10 is fine. Caught it early and the version of streamz with the breaking change hasn’t even been released yet

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by jdye64** [comment]
@bdice I don't have the permissions to alter the labels, hence `Label Checker` CI failure. Can you add those for me?

**Comment by vyasr** [comment]
Added labels. I assumed that this could be treated as a new feature even though technically we would be broken by the new version of the library.

**Comment by chinmaychandak** [comment]
Hey guys, would it be possible to merge this PR some time this week? This is blocking our cuStreamz dev work.

File: python/custreamz/custreamz/kafka.py [hunk 1]
```diff
@@ -270,3 +271,21 @@ def commit(self, offsets=None, asynchronous=True):
...
+        Consumes a single message, calls callbacks and returns events.
+
+        The application must check the returned Message object's
+        Message.error() method to distinguish between proper messages
+        (error() returns None), or an event or error
+        (see error().code() for specifics).
+
+        Parameters
+        ----------
+        timeout : float,
```
Comment (bdice) [inline python/custreamz/custreamz/kafka.py:286]:
```suggestion
        timeout : float
```

**Review by bdice — COMMENTED** [review]
@jdye64 Are there dependency changes needed to support this? Are these pinnings still accurate or do they need to be updated? https://github.com/rapidsai/cudf/blob/989c4116a3b054f46a7f1152f234b2601d2583b0/dependencies.yaml#L496-L508

**Comment by jdye64** [comment]
@bdice the 

> @jdye64 Are there dependency changes needed to support this? Are these pinnings still accurate or do they need to be updated?
> 
> https://github.com/rapidsai/cudf/blob/989c4116a3b054f46a7f1152f234b2601d2583b0/dependencies.yaml#L496-L508
> 
> https://github.com/rapidsai/cudf/blob/989c4116a3b054f46a7f1152f234b2601d2583b0/conda/recipes/custreamz/meta.yaml#L51

No, the dependencies are still fine as is. We didn't add a new function but rather just added a hook to call one that had always existed.

**Review by bdice — APPROVED** [review]
Looks fine to me. @jdye64 Do we need to add tests for this?

Do you have merge privileges? Do you want me to merge after CI passes?

**Pull request merged**